esmvalcore/preprocessor/_regrid.py

```diff
@@ -252,15 +252,64 @@ def regrid(cube, target_grid, scheme, lat_offset=True, lon_offset=True):
                 [coord] = coords
                 cube.remove_coord(coord)
 
-    # Perform the horizontal regridding.
-    if _attempt_irregular_regridding(cube, scheme):
-        cube = esmpy_regrid(cube, target_grid, scheme)
-    else:
-        cube = cube.regrid(target_grid, HORIZONTAL_SCHEMES[scheme])
+    # Return unaltered cube if hotizontal grid is the same.
+    if not _check_horiz_grid_closeness(cube, target_grid):
+
+        # Perform the horizontal regridding.
+        if _attempt_irregular_regridding(cube, scheme):
+            cube = esmpy_regrid(cube, target_grid, scheme)
+        else:
+            cube = cube.regrid(target_grid, HORIZONTAL_SCHEMES[scheme])
 
     return cube
 
 
+def _check_horiz_grid_closeness(cube1, cube2):
+    """
+    Check if two cubes have the same horizontal grid definition.
+
+    The result of the function is a boolean answer, if both cubes have the
+    same horizontal grid definition. The function checks both longitude and
+    latitude, based on extent and resolution.
+
+    Parameters
+    ----------
+    cube1 : cube
+        The first of the cubes to be checked.
+    cube2 : cube
+        The second of the cubes to be checked.
+
+    Returns
+    -------
+    bool
+
+    .. note::
+
+        The current implementation checks if the bounds and the
+        grid shapes are the same.
+        Exits on first difference.
+    """
+    # Go through the 2 expected horizontal coordinates longitude and latitude.
+    for coord in ['latitude', 'longitude']:
+        # Compare shapes.
+        if cube1.coord(coord).shape != cube2.coord(coord).shape:
+            # Different shapes detected.
+            return False
+        else:
+            # Iterate through the combined bounds.
+            for c1bndval, c2bndval in np.nditer(
```

Comment:
this function looks a bit more complicated than needed, I think it would be fine to just have
```python
coord1 = cube1.coord(coord)
coord2 = cube2.coord(coord)
return (coord1.shape == coord2.shape
        and np.allclose(coord1.points, coord2.points)
        and np.allclose(coord1.bounds, coord2.bounds))
```
because the coordinate arrays are typically much smaller than the data chunks anyway and we can probably rely on numpy to implement optimizations

Reply:
yeah, I was saying cube equality, that'll do straight out the box, minimal hassle

Reply:
As said before, I would not check points. I targeted the use case that the artificial target grid is the same as the dataset, but the coordinates are differently set. within the bounds.

Reply:
> this function looks a bit more complicated than needed, I think it would be fine to just have
> 
> ```python
> coord1 = cube1.coord(coord)
> coord2 = cube2.coord(coord)
> return (coord1.shape == coord2.shape
>         and np.allclose(coord1.points, coord2.points)
>         and np.allclose(coord1.bounds, coord2.bounds))
> ```
> 
> because the coordinate arrays are typically much smaller than the data chunks anyway and we can probably rely on numpy to implement optimizations

We have to only return True, if it is true for both horizontal dimensions. We still need an 'if', otherwise it would return true whenever the first one is ok.